### DISCUSSION TITLE
Example code demonstrate how to launch external process and manipulate it?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#244] [General]
I want to migrate some Python code to Objeck. The ability to launch external process and manipulate it is needed for this to be possible.

**Comment by objeck** [comment]
Refer to https://github.com/objeck/objeck-lang/blob/master/programs/tests/examples/some_examples.obs

**Reply by objeck** [reply]
Do you want to execute an external process or a thread?

**Reply by ghost** [reply]
An external process, e.g: `cmake`, `make`,...

**Comment by objeck** [comment]
Made updates to ``Runtime->CommandOutput(..)``. The new function returns a ``CommandOutput`` object with the process executed, status code, and output (as a string array).

**Reply by objeck** [reply]
You should be good now. We added an execution function to set environment parameters for the called process and then reset it, if needed, for the caller.

**Comment by ghost** [comment]
How to get the current working directory? On Python, it's simply `os.getcwd()`.

**Comment by objeck** [comment]
Which directory, install or working directory?

**Reply by ghost** [reply]
Please have a look at Python's `os.getcwd()`.

**Reply by ghost** [reply]
https://www.tutorialspoint.com/python/os_getcwd.htm

**Comment by objeck** [comment]
Thanks, good catch. Please refer to commit [fe4ddad](https://github.com/objeck/objeck-lang/commit/fe4ddadf125bfe35196182cd3ccfcbe2ab55b5df).

**Reply by objeck** [reply]
I will add a function to change the current working directory.

**Reply by ghost** [reply]
Please implement Python's `os.path.isdir()` too. Thank you.

**Comment by objeck** [comment]
Try ``System.IO.Filesystem.Directory->Exists()``

**Comment by ghost** [comment]
Please add method to delete directory. There is `File->Delete()` but no `Directory->Delete()`.

**Comment by objeck** [comment]
In commit ac71305
